pkg/apis/rook.io/v1/placement.go

```diff
@@ -45,19 +45,17 @@ func (p Placement) ApplyToPodSpec(t *v1.PodSpec, merge bool) {
 	}
 }
 
-func (p Placement) mergeNodeAffinity(nodeAffinity *v1.NodeAffinity, merge bool) *v1.NodeAffinity {
+func (p Placement) mergeNodeAffinity(nodeAffinity *v1.NodeAffinity) *v1.NodeAffinity {
 	// no node affinity is specified yet, so return the placement's nodeAffinity
 	result := p.NodeAffinity.DeepCopy()
 	if nodeAffinity == nil {
 		return result
 	}
 
 	// merge the preferred node affinity that was already specified, and the placement's nodeAffinity
-	if merge {
-		result.PreferredDuringSchedulingIgnoredDuringExecution = append(
-			nodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution,
-			p.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution...)
-	}
+	result.PreferredDuringSchedulingIgnoredDuringExecution = append(
```

Comment:
with this new change, won't be  merge `PreferredDuringSchedulingIgnoredDuringExecution`  from both `All` and `StorageClassDeviceSets`.  And end up having similar scenario mentioned in the original issue description but for `PreferredDuringSchedulingIgnoredDuringExecution`

Reply:
I believe the original concern was that "osd" placement should not ever be used for the storageClassDeviceSet placement, and vice versa. Merging the affinity with "all" for both types of OSDs seems like a good thing. Or which comment in the original issue are you referring to? I could certainly still be missing something.